### ISSUE TITLE
Не работает "вперед-назад" в браузере при аякс-фильтрации и пагинации.

### ISSUE THREAD

**Issue by f104** [bezumkin/mSearch2#32]
Вроде в js все window.history && history.pushState прописаны, но нет. Не работает. Даже на modx.pro

**Comment by soulcreate** [comment]
+1

**Comment by yujj** [comment]
Да, в историю push делается, а вот отслеживания изменения хэша нет, потому и не работает.

**Comment by yujj** [comment]
Вызов history.pushState(...) можно заменить на history.replaceState(...) в /assets/components/msearch2/js/web/default.js

т.е. 
```javascript
window.history.pushState(hash, '', document.location.pathname + hash);
```
на
```javascript
window.history.replaceState(hash, '', document.location.pathname + hash);
```
тогда переход по пагинации и фильтрация не будут добавляться в историю, но адрес будет меняться для возможности отправить ссылку на результат аякс-фильтрации/пагиции.

Либо нужно отслеживать изменения адреса:
```javascript
window.addEventListener('popstate', function(e){
 // код, который будет обрабатывать новый урл
}, false);
```

http://habrahabr.ru/post/123106/
